src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -137,215 +176,135 @@ public void AnalyzeCodeBlockEnd(CodeBlockAnalysisContext context)
                             RuleDoNotUseInsecureDTDProcessing,
                             env.XmlTextReaderDefinition.GetLocation(),
                             SecurityDiagnosticHelpers.GetLocalizableResourceString(
-                                nameof(DesktopAnalyzersResources.XmlTextReaderConstructedWithNoSecureResolutionMessage),
-                                env.EnclosingConstructSymbol.Name));
+                                nameof(DesktopAnalyzersResources.XmlTextReaderConstructedWithNoSecureResolutionMessage)
+                            )
+                        );
 
                         context.ReportDiagnostic(diag);
                     }
                 }
             }
 
-            public void AnalyzeNode(SyntaxNodeAnalysisContext context)
+
+            public void AnalyzeOperation(OperationAnalysisContext context)
             {
-                AnalyzeNodeForXmlDocument(context);
-                AnalyzeNodeForXmlTextReader(context);
-                AnalyzeNodeForXmlReaderSettings(context);
-                AnalyzeNodeForDtdProcessingOverloads(context);
-                AnalyzeNodeForDtdProcessingProperties(context);
+                switch (context.Operation.Kind)
+                {
+                    case OperationKind.ObjectCreationExpression:
+                        AnalyzeObjectCreationOperation(context);
+                        break;
+                    case OperationKind.AssignmentExpression:
+                        AnalyzeAssignment(context);
+                        break;
+                    case OperationKind.FieldInitializerAtDeclaration:
+                        AnalyzeFieldDeclaration(context);
+                        break;
+                    case OperationKind.VariableDeclaration:
+                        AnalyzeVariableDeclaration(context);
+                        break;
+                    case OperationKind.InvocationExpression:
+                        AnalyzeInvocation(context);
+                        break;
+                }
             }
 
-            private void AnalyzeNodeForDtdProcessingOverloads(SyntaxNodeAnalysisContext context)
+            private void AnalyzeInvocation(OperationAnalysisContext context)
             {
-                SyntaxNode node = context.Node;
-                SemanticModel model = context.SemanticModel;
+                IInvocationExpression invoke = context.Operation as IInvocationExpression;
```

Comment:
Rename to `invocationExpression` instead of `invoke`. With so much code, variable naming helps a lot